backend/app/routers/experiments.py

```diff
@@ -149,17 +155,78 @@ async def get_all_experiment_runs(pagination: Pagination = Depends()) -> Any:
 @router.get("/experiment-runs/{id}", response_model=ExperimentRunDetails | None)
 async def get_experiment_run(id: PydanticObjectId) -> Any:
     experiment_run = await ExperimentRun.get(id)
+    if experiment_run is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Specified experiment run doesn't exist",
+        )
     return experiment_run.map_to_detailed_response()
 
 
 @router.get("/experiment-runs/{id}/logs", response_class=PlainTextResponse)
 async def get_experiment_run_logs(id: PydanticObjectId) -> str:
-    # TODO
-    raise HTTPException(
-        status_code=status.HTTP_405_METHOD_NOT_ALLOWED, detail="Method not implemented"
+    experiment_run = await ExperimentRun.get(id)
+    if experiment_run is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Specified experiment run doesn't exist",
+        )
+    return experiment_run.logs
+
+
+@router.get("/experiment-runs/{id}/files/download", response_class=StreamingResponse)
+async def download_file(
+    id: PydanticObjectId,
+    filepath: str,
+    workflow_engine: WorkflowEngineBase = Depends(ReanaService.get_service),
+) -> bytes:
+    experiment_run = await ExperimentRun.get(id)
+    if experiment_run is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Specified experiment run doesn't exist",
+        )
+
+    tempdir = Path("./temp")
```

Comment:
I recommend using an already existent package for working with temporary directories so we don't need to manage anything. However, I am not sure how this works out with the StreamingResponse..
https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
I put as a TODO because I do remember that I was trying to use this package at first, but at the end I gave up on that idea

Reply:
OK, but please changed the hardcoded path `./temp` to point to the constant defined in `config.py` (as used in other parts of the code) 😉